### PR TITLE
337 Add validation for special chars in hdfs paths

### PR DESCRIPTION
**Pull request by Zejnilovic**
Added validation for HDFS paths so users don't input special characters or characters that may make troubles for us.

### CODE REVIEW

File: menas/ui/service/EntityValidationService.js [hunk 1]
```diff
@@ -17,26 +17,14 @@ jQuery.sap.require("sap.m.MessageBox");
...
-      sap.m.MessageToast.show("Please choose the Publish HDFS path of the " + sEntityType);
-    }
-
-    return isRawPathOk && isPublishPathOk;
-  };
-
-  this.hasValidHDFSPath = function (oEntity, sEntityType, oInput) {
-    let isOk = this._isValidHDFSPath(oEntity.hdfsPath, oInput);
+  this.hasValidHDFSPath = function (sHDFSPath, sEntityType, oInput) {
+    let rHDFSPathRegex = new RegExp("^[a-zA-Z0-9\-_\.\/=]+$");
```
Comment (GeorgiChochov) [inline menas/ui/service/EntityValidationService.js]:
```suggestion
    let rHDFSPathRegex = /^[\w\-\.\/=]+$/;
```

File: menas/ui/service/EntityDialog.js [hunk 2]
```diff
@@ -89,10 +89,18 @@ class DatasetDialog extends EntityDialog {
...
+      "Dataset Raw HDFS path",
+      this.oController.byId("selectedRawHDFSPathLabel"));
+    let hasValidPublishHDFSPath = EntityValidationService.hasValidHDFSPath(oDataset.hdfsPublishPath,
+      "Dataset publish HDFS path",
+      this.oController.byId("selectedPublishHDFSPathLabel"));
+    let hasExistingRawHDFSPath = hasValidRawHDFSPath ? this.oController.byId("newDatasetRawHDFSBrowser").validate() : false;
+    let hasExistingPublishHDFSPath = hasValidRawHDFSPath && hasValidPublishHDFSPath ?
+      this.oController.byId("newDatasetPublishHDFSBrowser").validate() : false;
+
+    return hasValidName && hasValidSchema && hasValidPublishHDFSPath && hasValidRawHDFSPath
```
Comment (GeorgiChochov) [inline menas/ui/service/EntityDialog.js]:
```suggestion
    return hasValidName && hasValidSchema
```
These are covered by `hasExistingRawHDFSPath` and `hasExistingPublishHDFSPath`, so they can be removed.

Comment (Zejnilovic) [reply menas/ui/service/EntityDialog.js]: Please check. I also fixed it on line `268`

File: menas/ui/service/EntityDialog.js [hunk 3]
```diff
@@ -253,9 +260,12 @@ class MappingTableDialog extends EntityDialog {
       this.oController.byId("newMappingTableName"));
     let hasValidSchema = EntityValidationService.hasValidSchema(oMT, "Mapping Table",
       this.oController.byId("schemaVersionSelect"));
-    let hasValidHDFSPath = this.oController.byId("addMtHDFSBrowser").validate();
+    let hasValidHDFSPath = EntityValidationService.hasValidHDFSPath(oMT.hdfsPath,
+      "Mapping Table HDFS path",
+      this.oController.byId("selectedHDFSPathLabel"));
+    let hasExistingRawHDFSPath = hasValidHDFSPath ? this.oController.byId("addMtHDFSBrowser").validate() : false;
```
Comment (GeorgiChochov) [inline menas/ui/service/EntityDialog.js]:
```suggestion
    let hasExistingHDFSPath = hasValidHDFSPath ? this.oController.byId("addMtHDFSBrowser").validate() : false;
```
`Raw` here is misleading for `MappingTables`

**Pull request merged**